components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java

```diff
@@ -172,4 +233,91 @@ public boolean terminateSessionBySessionId(String userId, String sessionId) {
         }
         return sessionsList;
     }
+
+    /**
+     * This method checks whether the user trying to terminate his own session. If the session does not owned by the
+     * user, then user must have "/permission/admin/manage/identity/authentication/session/delete" to terminate a
+     * session of another user.
+     *
+     * @param userId    id of the user
+     * @param sessionId id of the session
+     * @return true/false whether the user is authorized for the action
+     * @throws SessionManagementServerException if an error occurs while validating the user.
+     */
+    private boolean isUserAuthorized(String userId, String sessionId) throws SessionManagementServerException {
+
+        boolean isUserAuthorized;
+        try {
+            isUserAuthorized = UserSessionStore.getInstance().isExistingMapping(userId, sessionId);
+            if (!isUserAuthorized) {
+                isUserAuthorized = isActionForbidden();
+            }
+        } catch (UserSessionException | UserStoreException e) {
+            throw handleSessionManagementServerException(SessionMgtConstants.ErrorMessages
+                    .ERROR_CODE_UNABLE_TO_AUTHORIZE_USER, userId, e);
+        }
+        return isUserAuthorized;
+    }
+
+    /**
+     * This method checks whether the user trying to terminate his own session. If the session does not owned by the
+     * user, then user must have "/permission/admin/manage/identity/authentication/session/delete" to terminate a
+     * session of another user.
+     *
+     * @param user      user object
+     * @param idpId     id of the user's identity provider
+     * @param sessionId id of the session
+     * @return true/false whether the user is authorized for the action
+     * @throws SessionManagementServerException if an error occurs while validating the user.
+     */
+    private boolean isUserAuthorized(User user, int idpId, String sessionId) throws SessionManagementServerException {
+
+        boolean isUserAuthorized;
+        try {
+            isUserAuthorized = UserSessionStore.getInstance().isExistingMapping(user, idpId, sessionId);
+            if (!isUserAuthorized) {
+                isUserAuthorized = isActionForbidden();
+            }
+        } catch (UserSessionException | UserStoreException e) {
+            throw handleSessionManagementServerException(SessionMgtConstants.ErrorMessages
+                    .ERROR_CODE_UNABLE_TO_AUTHORIZE_USER, user.getUserName(), e);
+        }
+        return isUserAuthorized;
+    }
+
+    private boolean isActionForbidden() throws UserStoreException {
```

Comment:
Shall we change the method name to "isPermitted"

Reply:
this method is removed